src/transactions/transaction.service.ts

```diff
@@ -1,12 +1,33 @@
 import { ITransaction } from './models/Transactions';
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
 import { ConfigProvider } from 'src/system/ConfigProvider';
 import { TransactionGateway } from './transaction.gateway';
+import { ITransactionsListFilters } from 'src/requests/TransactionsListRequest';
 
 @Injectable()
 export class TransactionService {
 	constructor(private gateway: TransactionGateway, private configProvider: ConfigProvider) {
 	}
+	async filterTransactions(body: ITransactionsListFilters, userId: number): Promise<any> {
+		if (!userId) {
+			throw new NotFoundException('Invalid userId');
+		}
+		const filter: ITransactionsListFilters = {
+			orderBy: body.orderBy,
+			order: body.order,
+			from: body.from,
+			to: body.to,
+			accountId: body.accountId,
+			amountAbove: body.amountAbove,
+			amountBelow: body.amountBelow,
+			skip: body.skip,
+			limit: body.limit,
+		};
+		const result = await this.gateway.getTransactionsWithFilters(userId, filter);
+		const result2 = await this.gateway.getTransactionsCountWithFilters(userId, filter);
+		const res = [JSON.parse(JSON.stringify(result)), result2[0].count];
```

Comment:
You don't need to construct the filter, you already have it in body. Just send the body. Rename result2 to count and do [count] so u have the first item of the array. res is not an array, it's an object. const res = { data: result, metadata: {count: coumt.count}}